subsys/mgmt/mcumgr/smp.c

```diff
@@ -171,6 +173,55 @@ smp_rx_req(struct smp_transport *smpt, struct net_buf *nb)
 	k_work_submit_to_queue(&smp_work_queue, &smpt->work);
 }
 
+void smp_rx_clear(struct smp_transport *zst, bool clear_all, void *arg)
```

Comment:
Clear all is for future use, I've left it here so to avoid going through a stable API change. The usage would be for something like the UART transport, if a user wants to disable mcumgr then all pending queries can be cleared to use the UART for some other purpose